daffodil-core/src/main/scala/org/apache/daffodil/grammar/GrammarTerm.scala

```diff
@@ -94,16 +94,19 @@ abstract class Gram(contextArg: SchemaComponent)
    * Note: This should not evaluate the argument unless it has to.
    */
   def ~(qq: => Gram) = {
-    lazy val q = qq.deref
+    val q = qq.deref
```

Comment:
The comment above says this shouldn't evaluate qq unless it needs to. But this looks like it will always evaluate it now. But maybe it always did that regardless?

Reply:
Ah that was an oversight, from when i was trying to debug the dropped HGR. Will fix